pylabrobot/liquid_handling/liquid_handler.py

```diff
@@ -1383,6 +1399,21 @@ def save(self, path: str):
     with open(path, "w", encoding="utf-8") as f:
       json.dump(self.serialize(), f, indent=2)
 
+  def register_callback(self, method_name: str, callback: OperationCallback):
+    """Registers a callback for a specific method."""
+    if self._callbacks.get(method_name):
+      error_message = f"Callback already registered for: {method_name}"
+      raise RuntimeError(error_message)
+    self._callbacks[method_name] = callback
+
+  def _trigger_callback(self, method_name: str, *args, **kwargs):
+    """Triggers the callback associated with a method, if any."""
+    error = kwargs.pop("error", None)
+    if callback := self._callbacks.get(method_name):
+      callback(*args, error=error, **kwargs)
+    elif error:
```

Comment:
I generally try to explicitly use `is None` in PLR. Perhaps it's a personal quirk, but I like to be consistent :)

Reply:
I think `is None` is a totally Pythonic. It is late here, I'm missing something obvious :p

Reply:
Perhaps it's an issue on my side, but I still see `elif error:` :)

Reply:
I'm not confident that my update here is what you're looking for, let me know.